File: Wacomi.API/Models/Circles/CircleEvent.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Wacomi.API.Models
{
    public class CircleEvent : IDataItemWithMultiplePhotos, IAppUserLinkable
    {
        public int Id { get; set;}
        public Circle Circle{ get; set;}
        public int CircleId{ get; set;}
        [MaxLength(StaticData.ShortTextLength)]
        public string Title { get; set;}
        [MaxLength(StaticData.LongTextLength)]
        public string Description { get; set;}
        public bool IsPublic{ get; set;} = false;
        [Required]
        public DateTime FromDate{ get; set;}
        public DateTime? ToDate{ get; set;} = null;
        public string WebSiteUrls { get; set;}
        public int? MaxNumber{ get; set;}
        public bool ApprovalRequired{ get; set;} = false;
        public City City{ get; set;}
        public string Address{ get; set;}
        public int CityId{ get; set;}

        public ICollection<Photo> Photos {get; set;}
        public Photo MainPhoto {get; set;}
        public int? MainPhotoId {get; set;}
        public bool IsActive {get; set;}
        public DateTime DateCreated {get; set;} = DateTime.Now;
        public DateTime DateUpdated {get; set;} = DateTime.Now;
        public AppUser AppUser {get; set;}
        public int? AppUserId {get; set;}
        public virtual ICollection<CircleEventParticipation> CircleEventParticipations {get; set;}
    }
}

File: Wacomi.SPA/src/app/_models/CircleEvent.ts
import { IDataItemWithMultiplePhotos } from "./DataItemWithMultiplePhotos";
import { IAppUserLinkable } from "./IAppUserLinkable";
import { CircleEventParticipationStatus } from "./CircleEventParticipation";
import { Circle } from "./Circle";

// public int Id { get; set;}
//         public int CircleId{ get; set;}
//         public string Title { get; set;}
//         public string Description { get; set;}
//         public bool IsPublic{ get; set;} = false;
//         public string WebSiteUrls { get; set;}
//         public int? MaxNumber{ get; set;}
//         public bool ApprovalRequired{ get; set;} = false;
//         public string CityName { get; set;}
//         public int CityId{ get; set;}
//         public string Address{ get; set;}

//         public ICollection<PhotoForReturnDto> Photos {get; set;}
//         public PhotoForReturnDto MainPhoto {get; set;}
//         public int? MainPhotoId {get; set;}
//         public bool IsActive {get; set;}
//         public DateTime DateCreated {get; set;}
//         public DateTime DateUpdated {get; set;}
//         public AppUserForReturnDto AppUser {get; set;}
//         public int? AppUserId {get; set;}
//         public int NumberOfPaticipants{ get; set;}

export interface CircleEvent extends IDataItemWithMultiplePhotos, IAppUserLinkable {
    id: number;
    circleId: number;
    circle: Circle;
    title:string;
    description:string;
    isPublic:boolean;
    fromDate:string;
    toDate?:string;
    webSiteUrls:string;
    maxNumber?:number;
    approvalRequired: boolean;
    cityName: string;
    cityId: number;
    address: string;
    isActive:boolean;
    dateCreated:Date;
    dateUpdated:Date;
    numberOfPaticipants: number;
    numberOfWaiting: number;
    numberOfCanceled: number;
    
    myStatus?: CircleEventParticipationStatus;
    isCircleMember: boolean;
}
